### PR TITLE
Fix transfer encoding issue

### LINKED ISSUE #7408: Decoding mangled for some first message lines in preview
### Steps to reproduce

1.  Go to Mail app
2.  List messages (either on inbox or navigating to folder)
3.  See mangled preview

### Expected behavior

Correct preview decoding

### Actual behavior

Mangled preview / unintelligible on message list 

### Mail app version

1.14.0

### Mailserver or service

Dovecot

### Operating system

Debian 11.0

### PHP engine version

PHP 7.4

### Web server

Apache (supported)

### Database

MariaDB

### Additional info

It is a particular issue as this happens only for some specific messages from various senders (there are senders where preview is shown correctly), also, this was not happening some Mail versions ago.

Attached is a message source summary with the first 3 _actual_ message lines (all sensitive information has been redacted sans subject)

Also, a short capture on how the message preview on the message list is seen.

sshot_nexcloud_mail
[message_summary.txt](https://github.com/nextcloud/mail/files/9756094/message_summary.txt)

### PR DESCRIPTION
**Pull request by miaulalala**
Fixes #7408

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
If we reset `structure_analyzed` it will also fix existing data with wrong encoding

**Comment by miaulalala** [comment]
> If we reset `structure_analyzed` it will also fix existing data with wrong encoding

Repair step?

**Comment by miaulalala** [comment]
@ChristophWurst how would we handle this for backports? Is this an issue in Mail 2 only?

**Comment by ChristophWurst** [comment]
The feature exists since 1.14. You can take inspiration at https://github.com/nextcloud/mail/blob/main/lib/Migration/Version1140Date20220808203258.php#L44-L55 for resetting the flag.

Target branch updated because 2.1 has already been branched-off. Main will be 2.2.0

File: lib/Migration/Version1140Date20221027171138.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	public function __construct(MessageMapper $messageMapper,
+		LoggerInterface $logger) {
+		$this->logger = $logger;
+		$this->messageMapper = $messageMapper;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure(): ISchemaWrapper $schemaClosure
+g	 * @param array $options
```
Comment (ChristophWurst) [inline lib/Migration/Version1140Date20221027171138.php:50]: g

File: lib/Migration/Version1140Date20221027171138.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2022 Your name <your@email.com>
```
Comment (ChristophWurst) [inline lib/Migration/Version1140Date20221027171138.php:6]: :eyes:

File: lib/Migration/Version1140Date20221027171138.php (hunk 1 above)
Comment (miaulalala) [reply lib/Migration/Version1140Date20221027171138.php:50]: gg

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**